.github/workflows/test_on_push.yml

```diff
@@ -34,6 +34,8 @@ jobs:
 
   run_unit_integration_and_coverage_tests:
     needs: style
+    env:
+      BUILD_IDAKLU: OFF
```

Comment:
We don't install IDAKLU in tests because we don't install dependencies for Windows and Ubuntu doctests. To avoid this, we are skipping over IDAKLU tests in Windows in the develop branch. In this branch, we extended that list to Ubuntu and macOS.

Honestly, now that you say it, it was a dirty fix, and it's just better to skip the tests for Windows by setting `BUILD_IDAKLU=OFF` for it explicitly (otherwise, we need to install VCPKG packages and the tests would quickly go over 30+ minutes) and doctests (because we don't need IDAKLU over there anyways).

I will fix this part asap. Thanks for noticing!

Reply:
The problem with skipping the windows tests is that we won't know if there is a windows specific problem with IDAKLU. We want to make sure that the software works on all operating systems, we don't want to catch these problems only after we do a release....

Soon IDAKLU will be in its own repo and the process will get faster for PyBaMM, but until that time we are responsible for testing on windows as well.

Reply:
Okay, I will make the tests run on all platforms

Reply:
Yeah it is fine to skip it something specific like a docs build test, but for IDAKLU and other core functionality it should run on all platforms

Reply:
Yes, I agree. I know we did discuss it at length about how it doesn't help with CI time, but now that we're going to unvendor the solver, the circumstances have changed. We can set `BUILD_IDAKLU` to `ON`, even if we need to take a hit with longer running CI jobs – since it's going to be temporary.